### PR TITLE
Remove step map

### PR DESCRIPTION
**Pull request by oleurud**

### CODE REVIEW

File: lib/windshaft/backends/map.js [hunk 1]
```diff
@@ -39,77 +33,83 @@ MapBackend.prototype.createLayergroup = function(mapConfig, params, validatorMap
...
-            });
-        },
-        function finish(err) {
-            timer.end('createLayergroup');
-            callback(err, response, timer.getTimes());
+        return callback(null, layergroupData);
+    });
+};
+
+MapBackend.prototype._returnError = function(err, mapConfigId, callback) {
```
Comment (dgaubert) [inline lib/windshaft/backends/map.js]: what about `_deleteMapConfigAfterError` as a name for this method?

File: lib/windshaft/backends/map.js [hunk 2]
```diff
@@ -39,77 +33,83 @@ MapBackend.prototype.createLayergroup = function(mapConfig, params, validatorMap
...
+        if (err) {
+            return callback(err);
+        }
+
+        if (layersMetadata) {
+            layergroupData.metadata = {
+                layers: layersMetadata
+            };
+
+            // backwards compatibility for torque
```
Comment (dgaubert) [inline lib/windshaft/backends/map.js]: It's a good candidate to extract as a new method. isn't it?

File: lib/windshaft/backends/map.js [hunk 3]
```diff
@@ -39,77 +33,83 @@ MapBackend.prototype.createLayergroup = function(mapConfig, params, validatorMap
...
+        if (err) {
+            return this._returnError(err, mapConfig.id(), callback);
+        }
+
+        timer.end('createLayergroup');
+        return callback(null, layergroupData, timer.getTimes());
+    });
+};
+
+MapBackend.prototype._getLayergroupData = function(params, mapConfig, callback) {
```
Comment (dgaubert) [inline lib/windshaft/backends/map.js]: I'd reuse the old name for this method: `_fetchLayersMetadata` or `_getLayersMetadata`

File: lib/windshaft/backends/map.js [hunk 4]
```diff
@@ -39,77 +33,83 @@ MapBackend.prototype.createLayergroup = function(mapConfig, params, validatorMap
...
-                    }, {});
-                    if (Object.keys(torqueMetadata).length) {
-                        response.metadata.torque = torqueMetadata;
-                    }
+            return this._getDataAndFinish(params, mapConfig, timer, callback);
+        });
+    });
+};
+
+MapBackend.prototype._getDataAndFinish = function(params, mapConfig, timer, callback) {
```
Comment (dgaubert) [inline lib/windshaft/backends/map.js]: `_getDataAndFinish` =>`_getLayergroupMetadata`

File: lib/windshaft/backends/map.js (hunk 1 above)
Comment (oleurud) [reply lib/windshaft/backends/map.js]: I have changed 50 times all the names during the developing period and finally, I gave up. Thanks for the help

File: lib/windshaft/backends/map.js (hunk 2 above)
Comment (oleurud) [reply lib/windshaft/backends/map.js]: :+1:

**Pull request merged**